broker/broker_test.go

```diff
@@ -0,0 +1,174 @@
+package broker_test
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pivotal-cf/brokerapi"
+
+	"github.com/18F/cf-cdn-service-broker/broker"
+	"github.com/18F/cf-cdn-service-broker/models"
+)
+
+type FakeRouteManager struct {
+	RouteCreate models.Route
+	ErrorCreate error
+	RouteGet    models.Route
+	ErrorGet    error
+}
+
+func (m *FakeRouteManager) Create(instanceId, domain, origin string) (models.Route, error) {
+	return m.RouteCreate, m.ErrorCreate
+}
+
+func (m *FakeRouteManager) Get(instanceId string) (models.Route, error) {
+	return m.RouteGet, m.ErrorGet
+}
+
+func (m *FakeRouteManager) Update(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) Disable(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) Renew(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) RenewAll() {
+	return
+}
+
+func TestProvisioning(t *testing.T) {
+	suite.Run(t, new(ProvisionSuite))
+}
+
+type ProvisionSuite struct {
+	suite.Suite
+	Broker broker.CdnServiceBroker
+}
+
+func (s *ProvisionSuite) SetupTest() {
+	s.Broker = broker.CdnServiceBroker{
+		Manager: &FakeRouteManager{
+			RouteCreate: models.Route{State: "provisioned"},
```

Comment:
Not a blocker, but we might want to try representing the `State` as an enum.

http://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go